### ISSUE TITLE
GitVersion not working in Ubuntu 20.04

### ISSUE THREAD

**Issue by LTA-Thinking** [actions/virtual-environments#2293]
**Description**  
When the ubuntu-latest VM image switched to 20.04 we started getting the following error from GitVersion:
`System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: Unable to load shared library 'git2-572e4d8' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libgit2-572e4d8: cannot open shared object file: No such file or directory`

There was another error logged in the preceding step to install the .NET Core SDK:
`get-os-distro: Error: Distribution specific OS name and version could not be detected: UName = Linux`

This error didn't fail the install of .NET Core, but it seemed to make it so it didn't recognize what version of Linux it was running on. In the step's log it listed this as the version:
`Primary:linux-x64
Legacy:-x64`
Where in the previous build it listed this:
`Primary:linux-x64
Legacy:ubuntu.18.04-x64`

We are unsure if the two are connected, but changing the VM over to windows-latest fixed both issues.

**Area for Triage**:  
Unknown

**Question, Bug, or Feature?**:  
"Bug"

**Virtual environments affected**  
- [ ] Ubuntu 16.04
- [ ] Ubuntu 18.04
- [x] Ubuntu 20.04
- [ ] macOS 10.15
- [ ] macOS 11.0
- [ ] Windows Server 2016 R2
- [ ] Windows Server 2019

**Expected behavior**  
GitVersion runs successfully

**Actual behavior**  
Bug displayed above

**Repro steps**  
[Failed build](https://microsofthealthoss.visualstudio.com/FhirServer/_build/results?buildId=7642&view=logs&j=26d01f8a-cee5-51e1-6b92-33792f37eb80&t=f8d3df71-2120-562b-de25-9bed207ec714) - Only preserved for a week.

1. Have an azure devops environment with the GitVersion extension
2. Run this pipeline template (.yml file)
```
jobs:
- job: Semver
  pool:
    vmImage: 'ubuntu-latest'
  steps:
  - task: UseDotNet@2
    displayName: 'Use .NET Core sdk (for GitVersion)'
    inputs:
      packageType: sdk
      version: 2.1.x
	
  - task: GitVersion@5
    displayName: 'GitVersion'
    inputs:
      configFilePath: '$(Build.SourcesDirectory)/GitVersion.yml'
```

**Comment by foliveiraGH** [comment]
the same here!

**Comment by dmitry-shibanov** [comment]
Hello everyone, I think this issue related to [GitVersion](https://github.com/GitTools/GitVersion/issues/2471). It was fixed in new release (5.6.0) for ubuntu 20.04. You can try to use this snippet:
```
  - task: UseDotNet@2
    displayName: 'Use .NET Core sdk (for GitVersion)'
    inputs:
      packageType: sdk
      version: 5.x
  - task: gitversion/setup@0
    inputs:
      versionSpec: '5.6.0'
  - task: gitversion/execute@0
    inputs:
      useConfigFile: true
      configFilePath: '$(Build.SourcesDirectory)/GitVersion.yml'
```
Another way change ubuntu image to 18.04, because previously builds were run on ubuntu 18.04 and it worked.

**Comment by dmitry-shibanov** [comment]
Hello everyone. We're going to deploy new image with GitVersion (5.6.0) on January.

**Comment by foliveiraGH** [comment]
> Hello everyone, I think this issue related to [GitVersion](https://github.com/GitTools/GitVersion/issues/2471). It was fixed in new release (5.6.0) for ubuntu 20.04. You can try to use this snippet:
> 
> ```
>   - task: UseDotNet@2
>     displayName: 'Use .NET Core sdk (for GitVersion)'
>     inputs:
>       packageType: sdk
>       version: 5.x
>   - task: gitversion/setup@0
>     inputs:
>       versionSpec: '5.6.0'
>   - task: gitversion/execute@0
>     inputs:
>       useConfigFile: true
>       configFilePath: '$(Build.SourcesDirectory)/GitVersion.yml'
> ```
> 
> Another way change ubuntu image to 18.04, because previously builds were run on ubuntu 18.04 and it worked.

Hello Dmitry, Thanks a lot, I put the tasks as you suggested and It worked to me in Ubuntu 20.04

**Comment by dmitry-shibanov** [comment]
Hello everyone. Sorry for confusion, we do not preinstall `gitversion` for ubuntu. If you have any concerns feel free to reopen the issue.
